docs/contracts.rst

```diff
@@ -562,6 +562,8 @@ In addition to the list of state modifying statements explained above, the follo
         }
     }
 
+Pure functions are able to use the `revert()` and `require()` functions to revert potential state changes when an :ref:`error occurs <assert-and-require>`.
```

Comment:
Perhaps add: One could argue that reverting state changes actually performs state changes. This is not true, as only changes to the state made previously in parts of the code that did not have the `view` or `pure` restriction are undone and that code actually also has the option to catch the revert and not pass it on. This behaviour is also in line with the `staticcall` opcode.